tests/e2e/test_e2e.py

```diff
@@ -248,28 +178,21 @@ def test_e2e():
     )
 
     assert len(train_cands) == 2
-    assert len(train_cands[0]) == 3493
-    assert (
-        len(
-            candidate_extractor.get_candidates(
-                docs=[session.query(Document).filter(Document.name == "112823").first()]
-            )[0]
-        )
-        == 1432
-    )
+    assert len(candidate_extractor.get_candidates(docs)) == 2
 
     # Featurization
     featurizer = Featurizer(session, [PartTemp, PartVolt])
 
     # Test that FeatureKey is properly reset
     featurizer.apply(split=1, train=True, parallelism=PARALLEL)
     assert session.query(Feature).count() == 214
-    assert session.query(FeatureKey).count() == 1260
+    assert session.query(FeatureKey).count() == 1281
+    num_feature_keys = session.query(FeatureKey).count()
```

Comment:
I don't know if queries are cached, but perhaps assign the `num_feature_keys` first (1 query), and then assert `1281` on that so that there is only one query, not two.

Reply:
I don't know if queries are cached, either, but I resolved this issue in 7e6be6f34ff7efa0ad8a67b5134d0e318d851d80 by just reusing query results.